database/consensus.go

```diff
@@ -184,3 +193,145 @@ func (db *Db) GetGenesis() (*types.Genesis, error) {
 	row := rows[0]
 	return types.NewGenesis(row.ChainID, row.Time, row.InitialHeight), nil
 }
+
+// CheckIfBlockIsMissing checks if block is already stored in database
+func (db *Db) CheckIfBlockIsMissing(height int64) bool {
+	var block []dbtypes.BlockRow
+	stmt := `SELECT * FROM block WHERE height = $1`
+
+	err := db.Sqlx.Select(&block, stmt, height)
+
+	return err != nil
+}
```

Comment:
we should check len(block) instead of checking error, otherwise it would always return false (the err would still be nil if the selected row is empty)

Reply:
Ok I added the checks back to the code 👍🏻